ra/registration-authority.go

```diff
@@ -602,6 +608,11 @@ func (ra *RegistrationAuthorityImpl) UpdateRegistration(base core.Registration,
 
 // UpdateAuthorization updates an authorization with new values.
 func (ra *RegistrationAuthorityImpl) UpdateAuthorization(base core.Authorization, challengeIndex int, response core.Challenge) (authz core.Authorization, err error) {
+	// Refuse to update expired authorizations
+	if base.Expires == nil || base.Expires.Before(time.Now()) {
+		err = core.MalformedRequestError("Expired authorization")
```

Comment:
I think this should just be a plain error. MalformedRequestError implies we want to pass this through to the user as a 401. But really, this error should never happen, since we try to catch it early in WFE. So we would return a plain error, which would get turned into a 500 if it ever actually happened.

You might also be able to simplify the WFE logic by consolidating the expires logic in the RA, for both GetAuthorization and UpdateAuthorization. In that case you'd want to return core.GoneError here (which you'd have to create, and add to the rpcError switch in the RPC code), since you'd be indicating to the WFE what status code to return.